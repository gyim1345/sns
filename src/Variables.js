export const post = {
  like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
  tag: [('boat', 'sky', 'landscape')],
  _id: '5e8ec52d9bb4913b5a844423',
  id: 1,
  title: 'posting with id 1',
  imageUrl: 'http://localhost:3000/static/images/1.jpg',
  userName: 'gibong@gmail.com'
};

export const posts = [
  {
    like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
    tag: [('boat', 'sky', 'landscape')],
    _id: '5e8ec52d9bb4913b5a844423',
    id: 1,
    title: 'posting with id 1',
    imageUrl: 'http://localhost:3000/static/images/1.jpg',
    userName: 'gibong@gmail.com'
  }
];

export const currentUser = 'gibong@gmail.com';

export const currentUserName = 'gibong'

export const comments = [
  {
    like: [('gibong@gmail.com', 'guy@gmail.com', 'noone@gmail.com')],
    _id: '5e90119435a2d94f9e08e35a',
    id: 1,
    postLId: 1,
    title: 'title1',
    userName: 'gibong@gmail.com',
    replyToCommentId: null
  }
];

export const info = {
  userURL: 'http://localhost:3000/static/images/user1.png',
  name: 'gibong@gmail.com',
  postCount: 16,
  followerNumber: 3,
  nickName: '기봉이',
  introductory: '나는 기봉이'
};

export const sizeOfPicture = {
  width: '10px',
  height: '10px'
};
