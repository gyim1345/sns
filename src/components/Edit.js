import React, { useState } from "react";
import { editPostAPI } from "../apis/post";
import PropTypes from "prop-types";

function Edit({
  posting,
  setPosting,
  indexOfCommentOnThisPosting,
  currentUser,
  setCommentAPI
}) {
  const [edit, setEdit] = useState([]);
  const input = [];
  const index = useState(indexOfCommentOnThisPosting)
  const onEdit = e => {
    // edit[Id] = e.target.value;
    setEdit(e.target.value);
  };

  const onClick = async () => {
    try {
      const response = await editPostAPI(
        edit,
        posting,
        currentUser,
        indexOfCommentOnThisPosting
      );
      response.Message !== undefined
        ? alert(response.Message)
        : indexOfCommentOnThisPosting === undefined
        ? setPosting([response])
        : setCommentAPI(response);
      // setPosting([...posting, response])x
    } catch (e) {
      console.log(e);
    }
  };

  return (
    <>
      <input
        type="text"
        value={input[posting.id]}
        onChange={e => onEdit(e)}
        placeholder={posting.title || "edit comment"}
      />
      <button type="button" onClick={onClick} id="buttonEdit">
        Edit
      </button>
    </>
  );
}

Edit.propTypes = {
  currentUser: PropTypes.string,
  posting: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  setPosting: PropTypes.elementType,
  indexOfCommentOnThisPosting: PropTypes.number,
  setCommentAPI: PropTypes.elementType
};

Edit.defaultProps = {
  currentUser: "",
  posting: {},
  indexOfComment: undefined,
  cid: undefined
};

export default Edit;
