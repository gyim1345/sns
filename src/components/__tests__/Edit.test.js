import React from 'react';
import { mount } from 'enzyme';
import Edit from '../Edit';
import { post, currentUser } from '../../Variables';

describe('<Edit />', () => {
  it('renders <input>, <button>, and "Edit"', () => {
    const input = 'INPUT';
    const wrapper = mount(
      <Edit input={input} posting={post} currentUser={currentUser} />
    );

    expect(wrapper.props().posting).toBe(post);
    expect(wrapper.props().input).toBe(input);
    expect(wrapper.props().currentUser).toBe(currentUser);
    expect(wrapper.html()).toMatch('<input');
    expect(wrapper.html()).toMatch('<button');
    expect(wrapper.text()).toMatch('Edit');
    expect(wrapper).toMatchSnapshot();
  });
});
