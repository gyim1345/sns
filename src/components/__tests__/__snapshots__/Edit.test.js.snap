// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Edit /> renders <input>, <button>, and "Edit" 1`] = `
<Edit
  currentUser="gibong@gmail.com"
  input="INPUT"
  posting={
    Object {
      "_id": "5e8ec52d9bb4913b5a844423",
      "id": 1,
      "imageUrl": "http://localhost:3000/static/images/1.jpg",
      "like": Array [
        "gibong@gmail.com",
      ],
      "tag": Array [
        "landscape",
      ],
      "title": "posting with id 1",
      "userName": "gibong@gmail.com",
    }
  }
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Edit"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="button"
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkVkaXQuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBZ0NxQiIsImZpbGUiOiJFZGl0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0LCB7IHVzZVN0YXRlIH0gZnJvbSAncmVhY3QnO1xuaW1wb3J0IFN3YWwgZnJvbSAnc3dlZXRhbGVydDInO1xuaW1wb3J0IHsgY3NzIH0gZnJvbSAnQGVtb3Rpb24vY29yZSc7XG5cbmltcG9ydCB7IGVkaXRQb3N0QVBJIH0gZnJvbSAnLi4vYXBpcy9wb3N0JztcbmltcG9ydCBNb2RhbEJveEVkaXQgZnJvbSAnLi9Nb2RhbEJveEVkaXQnO1xuXG5mdW5jdGlvbiBFZGl0KHsgcG9zdGluZywgc2V0UG9zdGluZywgY3VycmVudFVzZXIsIHNldElzT3Blbk1vZGFsQm94LCBwb3N0aW5nQWxsIH0pIHtcbiAgY29uc3QgW2NsaWNrZWQsIHNldENsaWNrZWRdID0gdXNlU3RhdGUoZmFsc2UpO1xuXG4gIGNvbnN0IG9uQ2xpY2sgPSBhc3luYyAoKSA9PiB7XG4gICAgc2V0Q2xpY2tlZCh0cnVlKTtcbiAgfTtcblxuICByZXR1cm4gKFxuICAgIDw+XG4gICAgICB7Y2xpY2tlZCAmJiAoXG4gICAgICAgIDxNb2RhbEJveEVkaXRcbiAgICAgICAgICBjbGlja2VkPXtjbGlja2VkfVxuICAgICAgICAgIHBvc3Rpbmc9e3Bvc3Rpbmd9XG4gICAgICAgICAgc2V0UG9zdGluZz17c2V0UG9zdGluZ31cbiAgICAgICAgICBjdXJyZW50VXNlcj17Y3VycmVudFVzZXJ9XG4gICAgICAgICAgc2V0SXNPcGVuTW9kYWxCb3g9e3NldElzT3Blbk1vZGFsQm94fVxuICAgICAgICAgIHBvc3RpbmdBbGw9e3Bvc3RpbmdBbGx9XG4gICAgICAgIC8+XG4gICAgICApfVxuICAgICAgPGJ1dHRvbiB0eXBlPVwiYnV0dG9uXCIgb25DbGljaz17b25DbGlja30gaWQ9XCJidXR0b25FZGl0XCIgY3NzPXtidXR0b25Dc3N9PlxuICAgICAgICBFZGl0XG4gICAgICA8L2J1dHRvbj5cbiAgICA8Lz5cbiAgKTtcbn1cbmNvbnN0IGJ1dHRvbkNzcyA9IGNzc2BcbiAgYm9yZGVyOiAxcHggc29saWQgcmdiYSh2YXIoLS1kMGIsIDIxOSwgMjE5LCAyMTkpLCAxKTtcbiAgY29sb3I6IHJnYmEodmFyKC0tZjA3LCAzOCwgMzgsIDM4KSwgMSk7XG4gIGJhY2tncm91bmQtY29sb3I6IHdoaXRlO1xuICB3aWR0aDogMzAwcHg7XG4gIGhlaWdodDogNDhweDtcbiAgZm9udC1zaXplOiAxNHB4O1xuICBmb250LXdlaWdodDogYm9sZDtcbmA7XG5cbmNvbnN0IGJveEhpZGRlbiA9IGNzc2BcbiAgYm9yZGVyLXN0eWxlOiBoaWRkZW47XG4gIHBhZGRpbmctbGVmdDogMTJweDtcbiAgb3V0bGluZTogbm9uZTtcbmA7XG5cbmV4cG9ydCBkZWZhdWx0IEVkaXQ7XG4iXX0= */",
        "name": "18lsu5e-buttonCss",
        "styles": "border:1px solid rgba(var(--d0b,219,219,219),1);color:rgba(var(--f07,38,38,38),1);background-color:white;width:300px;height:48px;font-size:14px;font-weight:bold;;label:buttonCss;",
        "toString": [Function],
      }
    }
    id="buttonEdit"
    onClick={[Function]}
    type="button"
  >
    <button
      className="css-18lsu5e-buttonCss"
      id="buttonEdit"
      onClick={[Function]}
      type="button"
    >
      Edit
    </button>
  </EmotionCssPropInternal>
</Edit>
`;
